### ISSUE TITLE
Metrics/LineLength auto-correct breaks line in the wrong place

### ISSUE THREAD

**Issue by jonekdahl** [rubocop-hq/rubocop#7477]
Hi,

I have enabled auto-correction of `Metrics/LineLength` that was introduced in 0.68. I have also enabled the other [recommended cops](https://docs.rubocop.org/en/stable/cops_metrics/#metricslinelength) that were not enabled by default.

## Expected behavior

When I run auto-correct, the code should still be valid Ruby code.

## Actual behavior

In one instance, a line break was injected in the middle of an integer literal.

I have re-created the issue in a smaller example and this is the diff:

```diff
diff --git a/rubocop_line_length.rb b/rubocop_line_length.rb
index c505caa..01762f0 100644
--- a/rubocop_line_length.rb
+++ b/rubocop_line_length.rb
@@ -1,7 +1,9 @@
 class RubocopLineLength
   def breaks_this_incorrectly
     inventory_rows_to_fix = [
-      InventoryRow.new(year: 2019, week: 30, path: 'publisher_group:123;publisher:456;slot_type:midroll', to_book: 21906),
+      InventoryRow.new(year: 2019, week: 3
+0, path: 'publisher_group:123;
+publisher:456;slot_type:midroll', to_book: 21906),
       InventoryRow.new(year: 2019, week: 31, path: 'publisher_group:123;slot_type:preroll', to_book: 1558)
     ]
   end

```

## Steps to reproduce the problem

Create `.rubocop.yml`:

```
Metrics/LineLength:
  AutoCorrect: true

Layout/MultilineArrayLineBreaks:
  Enabled: true

Layout/MultilineHashKeyLineBreaks:
  Enabled: true

Layout/MultilineMethodArgumentLineBreaks:
  Enabled: true
```

Create `rubocop_line_length.rb`:

```ruby
class RubocopLineLength
  def breaks_this_incorrectly
    inventory_rows_to_fix = [
      InventoryRow.new(year: 2019, week: 30, path: 'publisher_group:123;publisher:456;slot_type:midroll', to_book: 21906),
      InventoryRow.new(year: 2019, week: 31, path: 'publisher_group:123;slot_type:preroll', to_book: 1558)
    ]
  end
end
```

Run `rubocop --only Layout,Metrics/LineLength --auto-correct rubocop_line_length.rb`

## RuboCop version

```
$ rubocop -V
0.76.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-linux)
```

**Comment by buehmann** [comment]
That's another off-by-one error in `Metrics/LineLength` (I recently fixed another one): https://github.com/rubocop-hq/rubocop/compare/master...buehmann:line-length/7477

The example reveals another bug. With the fix above, the cop autocorrects the lines to
```ruby
      InventoryRow.new(year: 2019, week: 30, path: 'publisher_group:123;
publisher:456;slot_type:midroll', to_book: 21906),
      InventoryRow.new(year: 2019, week: 31, path: 'publisher_group:123;
slot_type:preroll', to_book: 1558)
```
i.e. it inserts linebreaks into string literal (after the semicolons).

**Comment by maxh** [comment]
>inserts linebreaks into string literal

Here's a fix for that: https://github.com/rubocop-hq/rubocop/pull/7497
